### ISSUE TITLE
🐛 Bug: Incorrect information from `package-json/no-empty-fields` for `"browserslist": []`

### ISSUE THREAD

**Issue by anomiex** [JoshuaKGoldberg/eslint-plugin-package-json#1182]
### Bug Report Checklist

- [x] I have tried restarting my IDE and the issue persists.
- [x] I have pulled the latest `main` branch of the repository.
- [x] I have [searched for related issues](https://github.com/JoshuaKGoldberg/eslint-plugin-package-json/issues?q=is%3Aissue) and found none that matched my issue.

### Expected

`package-json/no-empty-fields` is correct in saying that having `"browserslist": []` in package.json does nothing and can be removed.

### Actual

Removing `"browserslist": []` from package.json results in different behavior, equivalent to having `"browserslist": [ "defaults" ]` in there.

### Reproduction

1. Create package.json with this content:
    ```json
    {
        "browserslist": []
    }
    ```
2. Create eslint.config.mjs with this content:
    ```js
    import packageJson from "eslint-plugin-package-json";

    export default [
            packageJson.configs.recommended,
    ];
    ```
3. Run `npm add browserslist eslint eslint-plugin-package-json`.
4. Run `npx browserslist`. Observe that there is no output from the command.
5. Run `npx eslint`. Observe that the output includes
   ```
     2:5  error  The field 'browserslist' does nothing and can be removed  package-json/no-empty-fields
   ```
6. As suggested, remove the field from package.json.
7. Run `npx browserslist` again. Observe that the command lists various browsers now.

You can also reproduce with something like
    ```json
        "browserslist": {
            "development": [],
            "production": [ "last 1 version", "> 1%", "not dead" ],
        }
    ```
in step 1 and running `BROWSERSLIST_ENV=development npx browserslist` in steps 4 and 7.

### Additional notes

It may be sufficient for the rule to have a configurable list of fields to ignore, rather than trying to have the rule know about every possible field that might treat an empty array or empty object differently from the field being unset.

**Comment by michaelfaith** [comment]
> It may be sufficient for the rule to have a configurable list of fields to ignore, rather than trying to have the rule know about every possible field that might treat an empty array or empty object differently from the field being unset.

This seems like a reasonable addition to the rule.  It's definitely not something the rule should have to manage on its own, especially since `browserslist` is a tool-specific concern and not part of the npm package.json spec.  We'd welcome a PR that adds an option to the rule for fields to ignore. 👍

**Comment by JoshuaKGoldberg** [comment]
@all-contributors please add @anomiex for bug.

> 🤖 Beep boop! This comment was added automatically by [all-contributors-auto-action](https://github.com/marketplace/actions/all-contributors-auto-action).
> Not all contributions can be detected from Git & GitHub alone. Please comment any missing contribution types this bot missed.
> ...and of course, thank you for contributing! 💙
